web_external/Viewer/index.jsx

```diff
@@ -170,5 +154,43 @@ class Viewer extends Component {
             </div>
         </div>
     }
+
+    getAnnotationForAFrame(frame) {
+        if (!this.props.annotationGeometryContainer ||
+            !this.props.annotationActivityContainer ||
+            !this.props.annotationTrackContainer) {
+            return;
+        }
+        var annotationGeometries = this.props.annotationGeometryContainer.getFrame(frame);
+        if (!annotationGeometries) {
+            return;
+        }
+        var data = annotationGeometries.filter((geometry) => {
+            return this.props.annotationTrackContainer.getEnableState(geometry.id1);
+        }).map((geometry) => {
+            var activities = this.props.annotationActivityContainer.getEnabledActivities(geometry.id1, frame);
+            var type = activities ? 'activity' : 'track';
+            return {
+                g0: geometry.g0,
+                type
```

Comment:
Okay, I guess this is JSX short-hand for `type: type`. Do we really need to use that here, though? The value is simple enough, I think it would be more clear to omit the temporary and just write out `type: activities ? 'activity' : 'track'`.